### ISSUE TITLE
OpenCV causes backpropagation to get stuck

### ISSUE THREAD

**Issue by MZSHAN** [pytorch/pytorch#63441]
## 🐛 Bug

I am trying to run the mnist script (https://github.com/pytorch/examples/blob/master/mnist/main.py) on a cpu.
Importing OpenCV in the script causes the script to get stuck indefinitely on loss.backward()
This happens even if opencv is not used anywhere in the script, importing it is causing this problem.

On stepping through the code using gdb, the stack trace is not visible. Instead of the current executing functions for each thread, gdb shows question marks.

## Environment

PyTorch version: 1.9.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-1054-aws-x86_64-with-glibc2.10
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.1
[pip3] sagemaker-pytorch-training==2.4.0
[pip3] torch==1.9.0
[pip3] torchvision==0.10.0
[conda] mkl                       2020.2                      256    anaconda
[conda] mkl-include               2020.2                      256    anaconda
[conda] numpy                     1.19.1           py38h30dfecb_0    anaconda
[conda] numpy-base                1.19.1           py38h75fe3a5_0    anaconda
[conda] sagemaker-pytorch-training 2.4.0                    pypi_0    pypi
[conda] torch                     1.9.0                    pypi_0    pypi
[conda] torchvision               0.10.0                   pypi_0    pypi
